asr1k_neutron_l3/models/neutron/l3/interface.py

```diff
@@ -125,8 +125,12 @@ def __init__(self, router_id, router_port, extra_atts):
         super(GatewayInterface, self).__init__(router_id, router_port, extra_atts)
 
         self.nat_address = self._nat_address()
+        
+        # annotate details about the router to the interface description so this can be picked up by SNMP
+        description = (f'type:gw;router:{self.router_id};network:{self.router_port["network_id"]};'
+                       f'subnet:{self._primary_subnet_id}')
```

Comment:
Have you looked into what the maximum string length for an interface description is or tested it?

Reply:
I did, it's 200 chars, which we are not hitting with these static lenght strings:
```
router(config-if)#description ?
  LINE  Up to 200 characters describing this interface
```